lensless/utils/dataset.py

```diff
@@ -1633,10 +1659,22 @@ def __getitem__(self, idx):
                 else:
                     return lensless, lensed, psf_aug, flip_lr, flip_ud
         else:
-            if not self.random_flip:
-                return lensless, lensed
-            else:
+            if self.random_flip:
                 return lensless, lensed, psf_aug, flip_lr, flip_ud
+            # Add background to achieve desired SNR
+            if self.bg is not None:
```

Comment:
this is where things get a bit tricky....

because we may want to return a background image in the other cases that involve returning a PSF and/or augmentation info.

How about something like:
```python
return_items = [lensless, lensed]
if self.multimask:
    if self.return_mask_label:
        return_items.append(mask_label)
    else:
        return_items.append(self.psf[mask_label])
    if self.random_flip:
        return_items.append(flip_lr)
        return_items.append(flip_ud)
else:
    if self.random_flip:
        return_items.append(psf_aug)
        return_items.append(flip_lr)
        return_items.append(flip_ud)

if self.bg is not None:
    # add simulated background to get image_with_bg and scaled_bg
    return_items[0] = image_with_bg
    return_items[0].append(scaled_bg)

return return_items
```

and then when indexing your dataset object
```python
return_items = dataset[0]
lensless = return_items[0]
lensed = return_items[1]
background = return_items[-1]
```